odoo/addons/base/models/ir_model.py

```diff
@@ -778,14 +778,14 @@ def _onchange_relation_table(self):
                     'message': _("The table %r if used for other, possibly incompatible fields.", self.relation_table),
                 }}
 
-    @api.onchange('required', 'ttype', 'on_delete')
-    def _onchange_required(self):
+    @api.constrains('ttype')
```

Comment:
Missing `'required'` and `'on_delete'`, right ?

Reply:
Right, sorry